### PR TITLE
add total number checkpoints to training scripts

### PR DESCRIPTION
**Pull request by williamberman**
re: https://github.com/huggingface/diffusers/issues/1297

### CODE REVIEW

File: examples/dreambooth/train_dreambooth.py [hunk 1]
```diff
@@ -195,6 +195,16 @@ def parse_args(input_args=None):
...
         ),
     )
+    parser.add_argument(
+        "--checkpointing_steps_total_limit",
+        type=int,
+        default=None,
+        help=(
+            "Max number of checkpoints to store. Passed as `total_limit` to the `Accelerator` `ProjectConfiguration`."
+            " See Accelerator::save_state https://huggingface.co/docs/accelerate/package_reference/accelerator#accelerate.Accelerator.save_state"
+            " for more docs"
```
Comment (sayakpaul) [inline examples/dreambooth/train_dreambooth.py:205]:
```suggestion
            " for more details"
```

File: examples/dreambooth/train_dreambooth.py [hunk 2]
```diff
@@ -488,11 +498,14 @@ def get_full_repo_name(model_id: str, organization: Optional[str] = None, token:
 def main(args):
     logging_dir = Path(args.output_dir, args.logging_dir)
 
+    accelerator_project_config = ProjectConfiguration(total_limit=args.checkpointing_steps_total_limit)
```
Comment (sayakpaul) [inline examples/dreambooth/train_dreambooth.py:501]: Nice! Didn't know about this. 

I always did the checkpointing logic manually with `accelerate` as shown in

https://github.com/huggingface/transformers/blob/9d1116e9951686f937d17697820117636bfc05a5/examples/pytorch/language-modeling/run_clm_no_trainer.py#L600

**Review by sayakpaul — APPROVED** [review]
Thanks!

File: examples/dreambooth/train_dreambooth.py (hunk 2 above)
Comment (pcuenca) [reply examples/dreambooth/train_dreambooth.py:501]: @sayakpaul it's pretty new, it was first made available in `accelerate 0.16.0` I think: https://github.com/huggingface/accelerate/pull/916

**Pull request merged**